### ISSUE TITLE
Run-AlCops/Extract-AppFileToFolder fail if runtime is >=8.0 and no ResourceExposurePolicy is set

### ISSUE THREAD

**Issue by marknitek** [microsoft/navcontainerhelper#2349]
**Describe the issue**
Run-AlCop and Extract-AppFileToFolder fail when the runtime is >=8.0 and no ResourceExposurePolicyis set. This is the case since the last addition to support ResourceExposurePolicy.

A empty Policy should be supported and therefore the could should check if the xml attributes are set before accessing these.
```
<Package>
  ...
  <ResourceExposurePolicy />
  <KeyVaultUrls />
</Package>
```

**Scripts used to create container and cause the issue**
```
Run-AlCop ...

or

Extract-AppFileToFolder -appFilename 'C:\Users\***\source\repos\LEP.Ext\project\Logico Solutions AG_LogiProjectmanagement_19.0.71.app' -generateAppJson -appFolder C:\temp\app

```

**Full output of scripts**
```
Extracting C:\ProgramData\BcContainerHelper\ca43d74b-b2a4-40f7-84e3-83e417844305\Logico Solutions AG_LogiProjectmanagement_19.0.71.app
Run-AlCops Telemetry Correlation Id: 8718fc58-7ad0-4c88-b100-4d36b6dacfb2
##[error]Cannot use previous app Logico Solutions AG_LogiProjectmanagement_19.0.71.app, it might be a runtime package.
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.3\AppHandling\Run-AlCops.ps1:107 char:17
...

Extract-AppFileToFolder -appFilename "C:\Users\***\Downloads\Logico%20Solutions%20AG_LogiProjectmanagement_19.0.71.0.app" -generateAppJson -appFolder C:\temp\app
Extracting C:\Users\***\Downloads\Logico%20Solutions%20AG_LogiProjectmanagement_19.0.71.0.app
Extract-AppFileToFolder Telemetry Correlation Id: e425045e-eae8-4f5a-9b97-2f9af6824e41
% : Die Eigenschaft "AllowDebugging" wurde für dieses Objekt nicht gefunden. Vergewissern Sie sich, dass die Eigenschaft vorhanden ist.
In C:\Program Files\WindowsPowerShell\Modules\bccontainerhelper\3.0.3\AppHandling\Extract-AppFileToFolder.ps1:147 Zeichen:100
+ ... ldNodes | Where-Object { $_.name -eq "ResourceExposurePolicy" } | % {
+                                                                       ~~~
    + CategoryInfo          : NotSpecified: (:) [ForEach-Object], PropertyNotFoundException
    + FullyQualifiedErrorId : PropertyNotFoundStrict,Microsoft.PowerShell.Commands.ForEachObjectCommand
```

**Screenshots**

![image](https://user-images.githubusercontent.com/2346803/158404396-605f0f52-5b6a-4c0a-858d-11aaffd924bd.png)

**Additional context**
- does it happen all the time?
- did it use to work?

The issue is introduced with bccontainerhelper 3.0.3

The issue lies in the newly added Code for the `Extract-AppFileToFolder` Function. The `AllowDebugging` Property/XmlAttribute (and others) are not set in this constellation and therefore an error occours about the Property that does not exist. 

>Die Eigenschaft "AllowDebugging" wurde für dieses Objekt nicht gefunden

```
$manifest.Package.ChildNodes | Where-Object { $_.name -eq "ResourceExposurePolicy" } | % { 

                $appJson += @{
                    "resourceExposurePolicy" = @{
                        "allowDebugging" = $_.AllowDebugging -eq "true"
                        "allowDownloadingSource" = $_.AllowDownloadingSource -eq "true"
                        "includeSourceInSymbolFile" = $_.IncludeSourceInSymbolFile -eq "true"
                    }
                }
            }
```
Unclear to me is why it actually fails, i suspect another issue in the weird XmlDocument handling in PowerShell (had many of these in the past). If i try to execute below code manually in powershell it works fine... 

Does it have to do with Strict-Mode (which is in the code but commented out?

**Comment by freddydk** [comment]
I have made the code more resilient, but I actually cannot repro the problem.
No matter what I insert as resourceExposurePolicy, I get all properties - set (defaulted to false).
Maybe your app was generated by a different compiler - but the next version will enumerate the attributes instead of just using the properties.

**Comment by freddydk** [comment]
Had to change this as apparently casing of properties is different in XML than in app.json.
New fix for the issue is in dev branch now

**Comment by freddydk** [comment]
fixed in 3.0.4
